app/services/preserved_object_handler.rb

```diff
@@ -183,37 +183,60 @@ def create_db_objects(status, validated=false)
         pc_attrs[:last_checked_on_storage] = t
       end
       PreservedCopy.create!(pc_attrs)
+    end
+
+    if transaction_results.empty?
       results << result_hash(CREATED_NEW_OBJECT)
+    else
+      results.concat(transaction_results)
```

Comment:
meh - I don't think this makes the code clearer, tbh.

Reply:
yeah, i'm not excited about the added complexity and decreased readability, but i do think it's slightly more correct behaviorally.